### PR TITLE
fix(vscode): preserve drafts across retry

### LINKED ISSUE #14290: Retry after a server error discards the unsent prompt from the chat box, with no way to recover it
### Cline Surface

VSCode Extension

### Cline Version

4.1.19 - installed on this machine, with 4.1.17 still present beside it. The build that was running when the
text was lost was not read before it was lost, so this is the installed version rather than a confirmed one.

Beta version: no.

### What happened

A long prompt was being typed into the chat box - several paragraphs of separate points, several minutes of
composition - while replies to earlier messages were still arriving in the same task. A server error ended the
attempt, the panel offered **Retry**, and pressing it left the chat box empty: the text that had been typed was
gone, and it was not anywhere in the task to copy back out of, so it had to be written again from memory.

What makes this worth a report rather than a shrug is which text was lost. Everything already sent is in the
task and can be scrolled back to; the draft had never been sent, so the box held the only copy of it. **Retry**
is also exactly the moment a user wants that copy back, because the natural next move is to send the same
message again.

### What could not be established

Whether the text was still in the box at the instant **Retry** was pressed, or the retry followed a send that
had already failed. Both end with an empty box and neither is recoverable, so the fix is the same either way,
but a maintainer reproducing this should know the reporter cannot now separate the two cases.

### What would fix it, from a user's side

Any of these closes it, and the first is what a user expects:

- Keep the composer's unsent text across the failure, so **Retry** retries the same message with the draft
  still in the box.
- Restore the draft into the box when the retry returns, and in the meantime show it somewhere it can be copied
  from.
- Keep an unsent draft in the task's own history, so it survives the error the way a sent message does, and a
  retry that is not wanted does not cost the typing.

### Impact

Low in frequency, high in cost when it lands: the loss is not an error message that can be retried, it is the
work itself, and the failure mode - retry - is the documented remedy for the error that causes it.

### PR DESCRIPTION
**Pull request by dominiccooney**
### Related Issue

**Issue:** #14290

### Description

VS Code now keeps unsent composer drafts across error recovery actions such as Retry.

Retry previously sent no composer content but still ran the same cleanup used by approval actions, deleting the only copy of text composed while a request was active. Footer actions also cleared text, quote context, images, and files separately, which made later message transitions prone to the same class of loss.

This change owns those values as one versioned draft. Only actions that submit the draft may consume it, and acknowledgement clears it only when no newer edit crossed the request boundary. Retry and Cancel cannot carry draft content; New Task drops only quote context tied to the old task.

Fixes #14290

### Test Procedure

- `cd apps/vscode/webview-ui && bun run test -- --reporter=dot`
- `cd apps/vscode && bun run check-types`
- `cd apps/vscode && bun run build:webview`
- In the debug host, trigger an OpenAI-compatible API failure while composing a draft; confirm Retry preserves it through a successful retried turn.
- Confirm normal follow-up submission clears the sent draft, Cancel preserves a streaming draft, Resume submits it, and the task-header New Task button keeps independent text available.

### Type of Change

-   [x] 🐛 Bug fix (non-breaking change which fixes an issue)
-   [ ] ✨ New feature (non-breaking change which adds functionality)
-   [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [x] ♻️ Refactor Changes
-   [ ] 💅 Cosmetic Changes
-   [ ] 📚 Documentation update
-   [ ] 🏃 Workflow Changes

### Pre-flight Checklist

-   [x] Changes are limited to a single feature, bugfix or chore (split larger changes into separate PRs)
-   [ ] Tests are passing (`bun test`) and code is formatted and linted (`bun run format && bun run lint`)
-   [x] I have reviewed [contributor guidelines](https://github.com/cline/cline/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Review by saoudrizwan — APPROVED** [review]

File: apps/vscode/webview-ui/src/components/chat/chat-view/hooks/useMessageHandlers.ts [hunk 1]
```diff
@@ -325,108 +328,58 @@ export function useMessageHandlers(messages: ClineMessage[], chatState: ChatStat
...
-						await TaskServiceClient.askResponse(
-							AskResponseRequest.create({
-								responseType: "yesButtonClicked",
-							}),
-						)
-					}
-					clearInputState()
+				case "proceed": {
+					const { draft } = invocation
+					const text = formatDraftText(draft.text.trim(), draft.activeQuote)
```
Comment (mkondratek) [inline apps/vscode/webview-ui/src/components/chat/chat-view/hooks/useMessageHandlers.ts:372]: `hasContent` is derived from the formatted text (after applying `formatDraftText`).

suggested fix: compute `hasContent` from `draft.text.trim()` before formatting; add a test covering an empty text and an active quote
